tests/actor/test_battery_status.py

```diff
@@ -597,6 +599,68 @@ async def test_sync_blocking_interrupted_with_invalid_message(
 
         await tracker.stop()
 
+    @time_machine.travel("2022-01-01 00:00 UTC", tick=False)
+    async def test_timers(
+        self, mock_microgrid: MockMicrogridClient, mocker: MockerFixture
+    ) -> None:
+        """Test if messages changes battery status/
+
+        Tests uses FakeSelect to test status in sync way.
+        Otherwise we would have lots of async calls and waiting.
+
+        Args:
+            mock_microgrid: mock_microgrid fixture
+        """
+        status_channel = Broadcast[Status]("battery_status")
+        request_result_channel = Broadcast[SetPowerResult]("request_result")
+
+        tracker = BatteryStatusTracker(
+            BATTERY_ID,
+            max_data_age_sec=5,
+            max_blocking_duration_sec=30,
+            status_sender=status_channel.new_sender(),
+            request_result_receiver=request_result_channel.new_receiver(),
+        )
+
+        battery_timer_spy = mocker.spy(tracker._battery.data_recv_timer, "reset")
+        inverter_timer_spy = mocker.spy(tracker._inverter.data_recv_timer, "reset")
+
+        assert tracker.battery_id == BATTERY_ID
+        assert tracker._last_status == Status.NOT_WORKING
+
+        select = FakeSelect(inverter=inverter_data(component_id=INVERTER_ID))
+        assert tracker._update_status(select) is None  # type: ignore[arg-type]
```

Comment:
I guess mypy only complains about `FakeSelect` or does it usually complain about any other select-type?

Reply:
It complains only about FakeSelect. It expect true Select from frequenz.channel repo.
But to test it easier and non-async way I created this FakeSelect.